elastic/elastic-agent
# Make delayed enrollment try indefinitely

## What does this PR do?



Make delayed enrollment try indefinitely to enroll into Fleet.

## Why is it important?



In some cases when a new machine comes online with delayed enrollment it doesn't have network access immediately so the enrollment needs to keep trying until it is able to successfully enroll into Fleet.

## Checklist




## Disruptive User Impact



Doesn't disrupt the user, it provides the more reliability for delayed enrollment.

## How to test this PR locally



1. Install with `--delay-enroll`
2. Manually start the Elastic Agent service
3. Should enroll and then run.

## Related issues


- Closes #4716
